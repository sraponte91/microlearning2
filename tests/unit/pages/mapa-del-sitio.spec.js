import { config, createLocalVue, mount } from '@vue/test-utils'
import VueMeta from 'vue-meta'
import MapaDelSitio from '@/pages/mapa-del-sitio.vue'
// create vue with vue-meta
const localVue = createLocalVue()
localVue.use(VueMeta, { keyName: 'head' })

const mock = {
    localVue,
    mocks: {}
}

const wrapper = mount(MapaDelSitio, mock)

describe('mapa-del-sitio.vue', () => {
    test('Is a Vue instance', () => {
        expect(wrapper.vm).toBeTruthy()
    })
    test('Renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    test('has correct <head> content', () => {
        expect(wrapper.vm.$metaInfo.title).toBe(config.mocks['$constants'].mapaDelSitio.meta.title + ' - ' +
            config.mocks['$constants'].home.meta.title)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'title'
        ).content).toBe(config.mocks['$constants'].mapaDelSitio.meta.metaTitle + ' - ' +
            config.mocks['$constants'].home.meta.title)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'description'
        ).content).toBe(config.mocks['$constants'].mapaDelSitio.meta.metaDescription)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'og:title'
        ).content).toBe(config.mocks['$constants'].mapaDelSitio.meta.ogTitle + ' - ' +
            config.mocks['$constants'].home.meta.title)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'og:description'
        ).content).toBe(config.mocks['$constants'].mapaDelSitio.meta.ogDescription)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'og:image'
        ).content).toBe(config.mocks['$constants'].baseURL + config.mocks['$constants'].mapaDelSitio.meta.ogImage)
    })
})
