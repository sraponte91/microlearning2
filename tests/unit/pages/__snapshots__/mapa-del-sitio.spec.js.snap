// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapa-del-sitio.vue Renders correctly 1`] = `
<main>
  <section
    class="sitemap"
  >
    <div
      class="container margin-0-auto"
    >
      <div
        class="row"
      >
        <div
          class="col-md-8 offset-md-2"
        >
          <h1
            class="text-center"
          >
            Términos y condiciones
          </h1>
           
          <ul>
            <li>
              <a
                href="#"
              >
                Nombre de página
              </a>
               
              <ul>
                <li>
                  <a
                    href="#"
                  >
                    Nombre de página
                  </a>
                </li>
                 
                <li>
                  <a
                    href="#"
                  >
                    Nombre de página
                  </a>
                </li>
                 
                <li>
                  <a
                    href="#"
                  >
                    Nombre de página
                  </a>
                   
                  <ul>
                    <li>
                      <a
                        href="#"
                      >
                        Nombre de página
                      </a>
                    </li>
                     
                    <li>
                      <a
                        href="#"
                      >
                        Nombre de página
                      </a>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
             
            <li>
              <a
                href="#"
              >
                Gestión de procesos
              </a>
            </li>
             
            <li>
              <a
                href="#"
              >
                Estrategia
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </section>
</main>
`;
